biz.aQute.bndlib/src/aQute/bnd/classfile/ClassFile.java

```diff
@@ -153,4 +166,28 @@ static ByteBuffer slice(DataInput in, int length) throws IOException {
 		in.readFully(array, 0, length);
 		return ByteBuffer.wrap(array, 0, length);
 	}
+
+	public byte[] toBytes() throws IOException {
+		ByteArrayOutputStream bout = new ByteArrayOutputStream();
+		DataOutputStream dout = new DataOutputStream(bout);
+		write(dout);
+		return bout.toByteArray();
+	}
+
+	public static ClassFile parseFile(File file) throws FileNotFoundException, IOException {
+		return parseInputStream(new FileInputStream(file));
```

Comment:
We try very hard to never use FileInputStream in Bnd for performance reasons. We generally use IO.stream.